evaluation/gaia/run_infer.py

```diff
@@ -0,0 +1,336 @@
+import asyncio
+import json
+import logging
+import multiprocessing as mp
+import os
+import pathlib
+import shutil
+import subprocess
+import time
+from concurrent.futures import ProcessPoolExecutor
+
+import huggingface_hub
+from datasets import load_dataset
+from tqdm import tqdm
+
+from evaluation.gaia.scorer import question_scorer
+from opendevin.controller.state.state import State
+from opendevin.core.config import config, get_llm_config_arg, get_parser
+from opendevin.core.logger import get_console_handler
+from opendevin.core.logger import opendevin_logger as logger
+from opendevin.core.main import main
+from opendevin.events.action import MessageAction
+from opendevin.events.serialization.event import event_to_dict
+from opendevin.runtime.docker.ssh_box import DockerSSHBox
+
+DATASET_CACHE_DIR = '~/.cache/open-devin/evals/gaia'
+DATASET_CACHE_DIR = os.path.expanduser(DATASET_CACHE_DIR)
+
+
+def cleanup():
+    print('Cleaning up child processes...')
+    for process in mp.active_children():
+        print(f'Terminating child process: {process.name}')
+        process.terminate()
+        process.join()
+
+
+def codeact_user_response(state: State) -> str:
+    msg = (
+        'Please continue working on the task on whatever approach you think is suitable.\n'
+        'If you think you have solved the task, please first send your answer to user through message and then exit.\n'
+        'IMPORTANT: YOU SHOULD NEVER ASK FOR HUMAN HELP.\n'
+    )
+    if state.history:
+        user_msgs = [
+            action
+            for action, _ in state.history
+            if isinstance(action, MessageAction) and action.source == 'user'
+        ]
+        if len(user_msgs) >= 2:
+            # let the agent know that it can give up when it has tried 3 times
+            return (
+                msg
+                + 'If you want to give up, run: <execute_bash> exit </execute_bash>.\n'
+            )
+    return msg
+
+
+def monologue_user_response(state: State) -> str:
+    raise NotImplementedError('MonologueAgent should never ask for user responses.')
+
+
+AGENT_CLS_TO_FAKE_USER_RESPONSE_FN = {
+    'CodeActAgent': codeact_user_response,
+    'MonologueAgent': monologue_user_response,
+}
+
+AGENT_CLS_TO_INST_SUFFIX = {
+    'CodeActAgent': 'When you think you have solved the question, please first send your answer to user through message and then exit.\n'
+}
+
+
+def process_instance(instance, agent_class, metadata, reset_logger: bool = True):
+    # create process-specific workspace dir
+    # we will create a workspace directory for EACH process
+    # so that different agent don't interfere with each other.
+    workspace_mount_path = os.path.join(config.workspace_mount_path, '_eval_workspace')
+    workspace_mount_path = os.path.join(workspace_mount_path, str(os.getpid()))
+    pathlib.Path(workspace_mount_path).mkdir(parents=True, exist_ok=True)
+    config.workspace_mount_path = workspace_mount_path
```

Comment:
Maybe you want to save the old `config.workspace_mount_path` as `old_workspace_mount_path`, and reset `config.workspace_mount_path = old_workspace_mount_path` at the end of this function?

I remember seeing the path eventually nested together as you process more instances, e.g., `_eval_workspace/111`, `_eval_workspace/111/_eval_workspace/112`, etc)